src/tools/immutable_object_cache/Types.cc

```diff
@@ -55,32 +41,33 @@ void ObjectCacheMsgData::decode(bufferlist& bl) {
 
 void ObjectCacheRequest::encode() {
   m_data.encode(m_data_buffer);
-  m_head.data_len = m_data_buffer.length();
-  ceph_assert(m_head_buffer.length() == 0);
-  m_head.encode(m_head_buffer);
-  ceph_assert(sizeof(ObjectCacheMsgHeader) == m_head_buffer.length());
 }
 
-bufferlist ObjectCacheRequest::get_head_buffer() {
-  return m_head_buffer;
+uint8_t get_header_size() {
+  //return sizeof(ObjectCacheMsgHeader);
+  return 6;
+}
```

Comment:
@dillaman I cleaned up the IPC message to use ENCODE/DECODE here. on read I just read 6 bytes first to get the msg content length then read the message from socket. Is this the correct way you mentioned earlier?

Reply:
I would still like to see the RPC messages broken out into individual message structures and not attempt to use one overly generic structure to represent all RPC messages.

Reply:
I see the gap now, will push a new commit for this, thanks!

Reply:
@dillaman I pushed a new commit 17e0504ed977dc1006ce3d7cb0c5113f5bdec78e to use specific message for different ops. The message reading still follow the origin way(read 6 bytes first, then read the whole message). On message decoding I will extract message type first then decode it based on the type. Is this close to your thoughts?

Reply:
Yeah, this is inline with what I was looking for. Thx.